.dotnet/src/Custom/Chat/ChatFunctionCall.cs

```diff
@@ -40,4 +40,24 @@ public partial class ChatFunctionCall
     /// </summary>
     [CodeGenMember("Arguments")]
     public BinaryData FunctionArguments { get; }
+
+    // CUSTOM: parameter order preserved.
+    public ChatFunctionCall(string functionName, BinaryData functionArguments)
```

Comment:
Should we add a small customization in the TypeSpec to change the parameter order there so that we don't lose support for generated code?

Reply:
I can do that. This might be the most subtle customization rule yet, but it works and the old order *was* better:

```yaml
  - remark: preserve earlier, more intuitive field order
    patterns:
      - ChatCompletionRequestAssistantMessage.function_call
      - ChatCompletionResponseMessage.function_call
    type: >-
      {
        name: string;
        arguments: string;
      }
    assert_existing_type: >-
      {
        arguments: string;
        name: string;
      }
```